src/NumericalAlgorithms/Spectral/Spectral.hpp

```diff
@@ -419,6 +419,31 @@ template <typename T>
 Matrix interpolation_matrix(const Mesh<1>& mesh,
                             const T& target_points) noexcept;
 
+/*!
+ * \brief Matrices that interpolation to the lower and upper boundaries of the
+ * element.
+ *
+ * Assumes that the logical coordinates are \f$[-1, 1]\f$. The first element of
+ * the pair interpolates to \f$\xi=-1\f$ and the second to \f$\xi=1\f$. These
+ * are just the Lagrange interpolating polynomials evaluated at \f$\xi=\pm1\f$.
+ * For Gauss-Lobatto points this the only non-zero element is at the boundaries
+ * and is one.
+ */
+const std::pair<Matrix, Matrix>& boundary_interpolation_matrix(
+    const Mesh<1>& mesh) noexcept;
+
+/*!
+ * \brief `DataVector`s of the Lagrange polynomials evaluated at the lower and
+ * upper boundaries divided by the volume quadrature weights.
+ *
+ * Assumes that the logical coordinates are \f$[-1, 1]\f$. The first element of
+ * the pair is the Lagrange polyonmials evaluated at \f$\xi=-1\f$ and the second
+ * at \f$\xi=1\f$. These are necessary for the lifting portion of a
+ * discontinuous Galerkin scheme that uses Gauss points.
+ */
+const std::pair<DataVector, DataVector>& boundary_lifting_term(
```

Comment:
If I understand correctly, using this function for lifting fluxes assumes mass-lumping, i.e. a diagonal mass matrix approximation. 1) It would be good to make that clear in the documentation and perhaps even the name of the function, and 2) since the lifting operation depends on a few choices (such as whether or not to use mass lumping, and whether or not to multiply the equations by the inverse mass at all) perhaps it would be better to keep the lifting utility in a separate file, e.g. `NumericalAlgorithms/DG/LiftFlux.hpp`. In fact `Spectral.hpp` is only very loosely tied to DG so far, so I think it's preferable to keep DG utility in `NumericalAlgorithms/DG`. What do you think?

Reply:
1) Yes, it assumes that but for Gauss points that is exact so there's no point in doing anything else. It only matters for Gauss-Lobatto.

2) mass lumping doesn't exist for Gauss points. I've never encountered anything where you don't multiply by the inverse mass matrix that's still called lifting. That wouldn't end up being a correction to the time derivatives.

I'm not sure why you say it's only loosely tied to DG. While basis functions are generic, they generic for nodal continuous and discontinuous Galerkin schemes. I'm not aware of any methods that are just special boundary conditions for these schemes. I'm also generally unhappy with `Spectral.hpp` but haven't yet figured out exactly what would be better. It's a bit long and messy, but if we start splitting some functions off into random places they will become very difficult to find. This is my main complaint about the current boundary scheme code, and so I'd rather not repeat that mistake by moving some functions into different directories.